RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -1,1028 +1,1149 @@
-			 << tz << " +/-" << setw(6) << sqrt(1./tks.dz2_[i]);
-
-			if (tks.tt[i]->track().quality(reco::TrackBase::highPurity)) {
-				std::cout  << " *";
-			} else {
-				std::cout  << "  ";
-			}
-			if (tks.tt[i]->track().hitPattern().hasValidHitInPixelLayer(PixelSubdetector::SubDetector::PixelBarrel, 1)) {
-				std::cout  << "+";
-			} else {
-				std::cout  << "-";
-			}
-			std::cout  << setw(1)
-			 << tks.tt[i]->track().hitPattern().pixelBarrelLayersWithMeasurement(); // see DataFormats/TrackReco/interface/HitPattern.h
-			std::cout  << setw(1)
-			 << tks.tt[i]->track().hitPattern().pixelEndcapLayersWithMeasurement();
-			std::cout  << setw(1) << hex
-					<< tks.tt[i]->track().hitPattern().trackerLayersWithMeasurement()
-					- tks.tt[i]->track().hitPattern().pixelLayersWithMeasurement()
-					<< dec;
-			std::cout  << "=" << setw(1) << hex
-					<< tks.tt[i]->track().hitPattern().numberOfHits(reco::HitPattern::MISSING_OUTER_HITS)
-					<< dec;
-
-			Measurement1D IP =
-					tks.tt[i]->stateAtBeamLine().transverseImpactParameter();
-			std::cout  << setw(8) << IP.value() << "+/-" << setw(6) << IP.error();
-			std::cout  << " " << setw(6) << setprecision(2)
-			 << tks.tt[i]->track().pt() * tks.tt[i]->track().charge();
-			std::cout  << " " << setw(5) << setprecision(2)
-			 << tks.tt[i]->track().phi() << " " << setw(5)
-			 << setprecision(2) << tks.tt[i]->track().eta();
-
-			double sump = 0.;
-			for (unsigned int ivertex = 0; ivertex < nv; ++ ivertex) {
-			  if  (std::fabs(y.z_[ivertex]-zdumpcenter_) > zdumpwidth_) continue;
-
-				if ((tks.pi_[i] > 0) && (tks.Z_sum_[i] > 0)) {
-					//double p=pik(beta,tks[i],*k);
-                                  double p = y.pk_[ivertex]  * exp(-beta * Eik(tks.z_[i], y.z_[ivertex], tks.dz2_[i],
-                                                                               tks.t_[i], y.t_[ivertex], tks.dt2_[i] )) / tks.Z_sum_[i];
-					if (p > 0.0001) {
-						std::cout  << setw(8) << setprecision(3) << p;
-					} else {
-						std::cout  << "    .   ";
-					}
-					E += p * Eik(tks.z_[i], y.z_[ivertex], tks.dz2_[i],
-                                                     tks.t_[i], y.t_[ivertex], tks.dt2_[i] );
-					sump += p;
-				} else {
-					std::cout  << "        ";
-				}
-			}
-			std::cout  << endl;
-		}
-		std::cout  << endl << "T=" << 1 / beta << " E=" << E << " n=" << y.getSize()
-			 << "  F= " << F << endl << "----------" << endl;
-	}
-#endif
-}
+#include "RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexException.h"
+
+#include <cmath>
+#include <cassert>
+#include <limits>
+#include <iomanip>
+#include "FWCore/Utilities/interface/isFinite.h"
+#include "vdt/vdtMath.h"
+
+using namespace std;
+#define VI_DEBUG

```

Comment:
This should get removed for standard usages

Reply:
Further: in the code you combine the usage of this macro and the "verbose_" flag: they mostly coincide, but not everywhere. Please review throughout the code and verify that they always correspond to what you want (e.g. check if you want to enclose within #ifdef VI_DEBUG every appearance of the "if (verbose_)" )